File: Contracts/src/main/java/org/cubexmc/contract/gui/ContractTerms.java
package org.cubexmc.contract.gui;

final class ContractTerms {
    private ContractTerms() {
    }

    static String preview(String description) {
        String clean = description == null ? "" : description.strip();
        if (clean.isEmpty()) {
            return "未填写";
        }
        clean = clean.replaceAll("\\R+", " / ");
        return clean.length() <= 48 ? clean : clean.substring(0, 45) + "...";
    }
}

File: Contracts/src/main/java/org/cubexmc/contract/gui/ContractTerms.kt
package org.cubexmc.contract.gui

internal object ContractTerms {
    @JvmStatic
    fun preview(description: String?): String {
        var clean = description?.trim() ?: ""
        if (clean.isEmpty()) {
            return "未填写"
        }
        clean = clean.replace(Regex("\\R+"), " / ")
        return if (clean.length <= 48) clean else clean.substring(0, 45) + "..."
    }
}
